plugins/modules/ecs_taskdefinition.py

```diff
@@ -366,6 +792,37 @@ def main():
         if launch_type == 'FARGATE' and network_mode != 'awsvpc':
             module.fail_json(msg="To use FARGATE launch type, network_mode must be awsvpc")
 
+        for container in module.params['containers']:
+            if container.get('links') and network_mode != 'bridge':
+                module.fail_json(msg='links parameter is only supported if the network mode of a task definition is bridge.')
+
+            for environment in container.get('environment', []):
+                environment['value'] = to_text(environment['value'])
+
+            for environment_file in container.get('environmentFiles', []):
+                if environment_file['value'] != 's3':
+                    module.fail_json(msg='The only supported value for the file type is s3.')
+
+            if container.get('maxSwap') and launch_type == 'FARGATE':
+                module.fail_json(msg='maxSwap parameter is only supported withFargate launch type.')
```

Comment:
```suggestion
            if container.get('maxSwap') and launch_type == 'FARGATE':
                module.fail_json(msg='maxSwap parameter is only supported with Fargate launch type.')
```

I presume there's a missing space here.

Is this logic backwards?  You're using `launch_type == 'FARGATE'` and then say "only supported with Fargate"

Reply:
Docs seem to imply "maxSwap parameter is not supported with Fargate launch type"

Reply:
You're right, the control logic was correct, but the message I was printing was showing backward logic. I think I've done some copy/paste and forgot to revise the sentence correctly. My bad, sorry! I solved it for this and the other cases as well. Thank you.